Sources/MongoSwift/ReadPreference.swift

```diff
@@ -0,0 +1,167 @@
+import libmongoc
+
+/// A class to represent a MongoDB read preference.
+final public class ReadPreference {
+
+    /// An enumeration of possible ReadPreference modes.
+    public enum Mode: String {
+        /// See https://docs.mongodb.com/manual/reference/read-preference/
+        case primary
+        case primaryPreferred
+        case secondary
+        case secondaryPreferred
+        case nearest
+
+        internal var readMode: mongoc_read_mode_t {
+            switch self {
+            case .primary:
+                return MONGOC_READ_PRIMARY
+            case .primaryPreferred:
+                return MONGOC_READ_PRIMARY_PREFERRED
+            case .secondary:
+                return MONGOC_READ_SECONDARY
+            case .secondaryPreferred:
+                return MONGOC_READ_SECONDARY_PREFERRED
+            case .nearest:
+                return MONGOC_READ_NEAREST
+            }
+        }
+
+        internal init? (readMode: mongoc_read_mode_t) {
+            switch readMode {
+            case MONGOC_READ_PRIMARY:
+                self = .primary
+            case MONGOC_READ_PRIMARY_PREFERRED:
+                self = .primaryPreferred
+            case MONGOC_READ_SECONDARY:
+                self = .secondary
+            case MONGOC_READ_SECONDARY_PREFERRED:
+                self = .secondaryPreferred
+            case MONGOC_READ_NEAREST:
+                self = .nearest
+            default:
+                return nil
```

Comment:
This was necessary since Swift doesn't translate a C enum directly to a Swift enum, so we have to refer to the C enum values as constants. That said, `nil` return value here will translate to a `preconditionFailure` in the `mode` accessor.

Reply:
If `///` is for documentation and this is internal, wouldn't a non-documentation comment would be preferred?

Reply:
I've opted (no pun intended) to move the precondition failure here, since we only call this once from the `mode` accessor. That let's us remove the optional behavior of this initializer and simplifies the accessor.

Reply:
jazzy decides what to document based on access control level - it defaults to a minimum ACL of public, so private and internal things are skipped. 

so I usually will use documentation comments on internal stuff anyway to be consistent stylistically (and it would be convenient if we decided we wanted to generate a separate docset of our internal types in the future.) 

in any case, that sounds reasonable to me!